### ISSUE TITLE
Optional fields

### ISSUE THREAD

**Issue by rendaw** [marshallpierce/base64-serde#16]
Sorry for another "this probably isn't the right place for it but this is the only place for it right now" type of ticket.

So if you have a field:
```
#[serde(with = "serdeb64")]
x: Option<Vec<u8>>,
```

This doesn't work.  The workaround is to implement a wrapper type like:
```
pub struct ConfigurationBytes(Vec<u8>);

impl Serialize for ConfigurationBytes {
    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error> where
        S: Serializer {
        SerdeB64::serialize(&self.0, serializer)
    }
}

impl<'de> Deserialize<'de> for ConfigurationBytes {
    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error> where
        D: Deserializer<'de> {
        SerdeB64::deserialize(deserializer).map(|v| ConfigurationBytes(v))
    }
}

impl Debug for ConfigurationBytes {
    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
        self.0.fmt(f)
    }
}

impl From<&[u8]> for ConfigurationBytes {
    fn from(v: &[u8]) -> Self {
        ConfigurationBytes(v.to_vec())
    }
}

impl From<Vec<u8>> for ConfigurationBytes {
    fn from(v: Vec<u8>) -> Self {
        ConfigurationBytes(v)
    }
}
```

But it's a lot of boilerplate of course (plus `.into()`, `.0`, etc).

Handling every combination of `Option` `Box` `Rc` here would probably be crazy but if might be nice to have `Option` support as well.

**Comment by marshallpierce** [comment]
Not out of the question to do it here but wouldn't that mean every format like base64 would need to do it, eventually? I'm kinda surprised serde doesn't Do The Right Thing(TM) here.

**Comment by rendaw** [comment]
Yeah, I looked for a way to add a custom serializer globally or at least globally within the struct but I didn't find anything that looked relevant.  I think you'd need sub-field annotations somehow to say "use base64 for the first generic type" or similar...

**Comment by marshallpierce** [comment]
That's a bummer. I'll get to it eventually but it's probably gonna take me a while to bubble up to the top of my OSS todo.

**Comment by rendaw** [comment]
Okay, after talking about it a bit maybe it would be better if I mention this in serde proper and see if there's a better solution.

**Comment by rendaw** [comment]
Well at least confirmation: https://github.com/serde-rs/serde/issues/723

**Comment by rendaw** [comment]
In any case I only opened this issue so I can safely forget everything, so no hurry at all.
